x-pack/plugins/integrations_manager/README.md

```diff
@@ -0,0 +1,39 @@
+# Integrations Manager
+
+## Directory overview
+```
+x-pack/plugins/integrations_manager/
+├── common 
+│   ├── constants.ts        # common strings, mostly. id, paths, etc
+│   └── types.ts            # as we find common types
+├── index.ts                # legacy plugin definition
+├── kibana.json             # new platform style shown in MIGRATION.md
+├── public
+│   ├── components          # plugin-specific components (used by screens/*)
+│   ├── data.ts             # all client-side data fetching
+│   ├── index.tsx           # React root. Client-side routes applied
+│   ├── register_feature.ts # add to registry for /#/home/feature_directory
+│   ├── routes.tsx          # every re: client-side routes. paths, mappings
+│   └── screens             # typically a view for a client-side route
+│       ├── detail.tsx      # integration detail view i.e. '/detail/:pkgkey'
+│       └── home.tsx        # integration list view i.e. '/'
+└── server            
+    ├── index.ts            # new platform style shown in MIGRATION.md 
+    ├── plugin.ts           # new platform style shown in MIGRATION.md
+    ├── registry.ts         # anything registry-related
+    ├── routes.ts           # every re: server-side routes. paths, mappings
+    └── saved_objects.ts    # anything saved objects-related
+```
```

Comment:
Wow, awesome detail here! In general, I think this kind of documentation is the kind that is really hard to maintain and goes out of date, when the file structure is somewhat self-documenting already. Maybe we can add README files in the directories that need some extra explanation?

Reply:
There's always the threat of doc rot and drifting away from the code. I'll update as it changes and move to something else (inline docs, per-directory README, etc) as needed.